client/src/main/com/sinch/sdk/domains/voice/models/svaml/MenuOptionAction.java

```diff
@@ -0,0 +1,30 @@
+package com.sinch.sdk.domains.voice.models.svaml;
+
+/** Navigates to the named menu */
+public class MenuOptionAction {
+
+  private final MenuOptionActionType type;
+  private final String id;
+
+  private MenuOptionAction(MenuOptionActionType type, String id) {
+    this.type = type;
+    this.id = id;
+  }
+
+  public String getId() {
+    return id;
+  }
+
+  public MenuOptionActionType getType() {
+    return type;
+  }
+
+  public static MenuOptionAction from(MenuOptionActionType type, String id) {
```

Comment:
Usually you are using the builder pattern to construct an Object but here you are using a constructor look-alike. I'm curious about the reason

Reply:
Yes, mainly governed by usage simplification

But in the future, if this choice was a bad one; we could move back to builder pattern ... an deprecate current static factory.